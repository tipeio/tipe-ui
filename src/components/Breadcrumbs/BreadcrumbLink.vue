<template>
  <button
    :data-tipe-ui="$options.name"
    @click="$emit('click', $event)"
  >
    <div 
      v-if="icon" 
      class="icon-container">
      <tipe-icon :icon="icon"/>
    </div>
    <div class="label">
      {{ label }}
    </div>
  </button>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '@/components/Icon'

export default {
  name: 'TipeBreadcrumbLink',
  components: {
    TipeIcon
  },
  props: {
    icon: vueTypes.string.def(''),
    label: vueTypes.string.def(''),
    name: vueTypes.string.def('')
  }
}
</script>

<style lang="postcss" scoped>
button {
  display: flex;
  flex-direction: row;
  height: 2.125rem;
  width: auto;
  padding-right: 0.625rem;
  border: 1px solid var(--gray-blue-light);
  border-radius: 0.1875rem;
  box-sizing: border-box;
  text-decoration: none;
  font-size: 0.8125rem;
  background-color: #fff;
}

.icon-container {
  height: 100%;
  width: 1.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
}

.label {
  padding-left: 0.5rem;
}
</style>
