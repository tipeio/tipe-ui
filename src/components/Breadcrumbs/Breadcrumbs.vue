<template>
  <nav :data-tipe-ui="$options.name">
    <tipe-breadcrumb-link
      v-for="(link, i) in links"
      :key="i"
      v-bind="link"
      @click="$emit('click', link.name)"
    />
  </nav>
</template>

<script>
import vueTypes from 'vue-types'
import TipeBreadcrumbLink from './BreadcrumbLink'

export default {
  name: 'TipeBreadcrumbs',
  components: {
    TipeBreadcrumbLink
  },
  props: {
    links: vueTypes.arrayOf(
      vueTypes.shape({
        icon: vueTypes.string.def(''),
        label: vueTypes.string.def(''),
        name: vueTypes.string.def('')
      })
    )
  }
}
</script>

<style lang="postcss" scoped>
nav {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: row;
}

button:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
  border-left: none;
}

button:not(:last-child) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
</style>
