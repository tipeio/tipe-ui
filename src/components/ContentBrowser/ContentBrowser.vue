<template>
  <section :data-tipe-ui="$options.name">
    <header class="nav-container">
      <tipe-breadcrumbs :links="breadcrumbLinks" />
    </header>
    <div class="content-window">
      <div class="scrollable">
        <div class="browser">
          <section class="finder-container">
            <header class="finder-header">Folders</header>
            <tipe-finder />
          </section>
          <section class="finder-container">
            <header class="finder-header">Documents</header>
            <tipe-finder />
          </section>
        </div>
      </div>
    </div>
  </section>
</template>

<script>
import vueTypes from 'vue-types'
import LinkShape from '@/types/Link'
import TipeFinder from '@/components/Finder'
import TipeBreadcrumbs from '@/components/Breadcrumbs'

export default {
  name: 'TipeContentBrowser',
  components: {
    TipeFinder,
    TipeBreadcrumbs
  },
  props: {
    breadcrumbLinks: vueTypes.arrayOf(LinkShape)
  }
}
</script>


<style lang="postcss" scoped>
[data-tipe-ui='TipeContentBrowser'] {
  display: grid;
  grid-template-rows: 3.3125rem 1fr;
  height: 100%;
  width: 100%;
}

.nav-container {
  background-color: var(--gray-lightest);
}

.content-window {
  position: relative;
  overflow: hidden;
}

.scrollable {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  overflow-y: scroll;
  padding-right: 17px;
}

.browser {
  display: flex;
  flex-direction: column;
}

.finder-container {
  padding-top: 3rem;
  padding-bottom: 1rem;
  box-sizing: border-box;
}

.finder-header {
  font-size: 1.3125rem;
  font-weight: 600;
  height: 3.5rem;
}
</style>
