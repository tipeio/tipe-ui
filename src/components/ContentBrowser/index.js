export { default } from './ContentBrowser.vue'
