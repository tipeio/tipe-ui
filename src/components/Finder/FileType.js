import vueTypes from 'vue-types'

const fileTypes = ['document', 'folder']

export default vueTypes.shape({
  label: vueTypes.string,
  type: vueTypes.oneOf(fileTypes)
})
