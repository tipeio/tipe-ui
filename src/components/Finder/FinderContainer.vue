<template>
  <type-finder-empty v-if="empty" />
  <type-finder-viewer
    v-else
    :files="files"
    :layout="layout"
    :icon="icon"
    @select="file => $emit('select', file)"
  />
</template>

<script>
import vueTypes from 'vue-types'
import interfaces from '@tipe/tipe-interfaces'
import TypeFinderViewer from './FinderViewer.vue'
import TypeFinderEmpty from './FinderEmpty.vue'

export default {
  name: 'TipeFinderContainer',
  components: {
    TypeFinderViewer,
    TypeFinderEmpty
  },
  props: {
    files: vueTypes.arrayOf(vueTypes.shape(interfaces.file)).def([]),
    layout: vueTypes.oneOf(['list', 'grid']).def('list'),
    icon: vueTypes.oneOf(['rectangle', 'square', 'details']).def('rectangle')
  },
  computed: {
    empty() {
      return this.files.length < 1
    }
  }
}
</script>
