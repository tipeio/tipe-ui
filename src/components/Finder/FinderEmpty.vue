<template>
  <div data-tipe-ui="TipeFinder">Empty</div>
</template>

<style lang="postcss" scoped>
[data-tipe-ui='TipeFinder'] {
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 0.8125rem;
  color: var(--text-gray-normal);
  background-color: var(--gray-lightest);
}
</style>
