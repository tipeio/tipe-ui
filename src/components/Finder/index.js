export { default } from './FinderContainer.vue'
