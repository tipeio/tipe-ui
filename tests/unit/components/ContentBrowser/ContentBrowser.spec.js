import { shallowMount } from '@vue/test-utils'
import ContentBrowser from '@/components/ContentBrowser'
import Breadcrumbs from '@/components/Breadcrumbs'
import Finder from '@/components/Finder'
import linkMock from '@/mocks/Link'
import fileMock from '@/mocks/File'

describe('ContentBrowser', () => {
  describe('<ContentBrowser>', () => {
    it('matches previous snapshot', () => {
      const wrapper = shallowMount(ContentBrowser)

      expect(wrapper).toMatchSnapshot()
    })

    it('has correct data-tipe-iu attibute', () => {
      const wrapper = shallowMount(ContentBrowser)

      expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeContentBrowser')
    })
  })

  describe(':props', () => {
    it(':breadcrumbsLinks - should pass breadcrumbsLinks prop to the Breadcrumbs component', () => {
      const propsData = { breadcrumbLinks: [linkMock()] }
      const wrapper = shallowMount(ContentBrowser, { propsData })

      expect(wrapper.find(Breadcrumbs).props().links).toEqual(
        propsData.breadcrumbLinks
      )
    })

    it(':documents - should pass folders prop to the first Finder component', () => {
      const propsData = { folders: [fileMock({ type: 'folder' })] }
      const wrapper = shallowMount(ContentBrowser, { propsData })

      expect(
        wrapper
          .findAll(Finder)
          .at(0)
          .props().files
      ).toEqual(propsData.folders)
    })

    it(':documents - should pass documents prop to the second Finder component', () => {
      const propsData = { documents: [fileMock({ type: 'document' })] }
      const wrapper = shallowMount(ContentBrowser, { propsData })

      expect(
        wrapper
          .findAll(Finder)
          .at(1)
          .props().files
      ).toEqual(propsData.documents)
    })
  })
})
