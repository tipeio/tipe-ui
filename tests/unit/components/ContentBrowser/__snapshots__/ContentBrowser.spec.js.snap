// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBrowser <ContentBrowser> matches previous snapshot 1`] = `
<section data-tipe-ui="TipeContentBrowser">
  <header class="nav-container">
    <TipeBreadcrumbs-stub></TipeBreadcrumbs-stub>
  </header>
  <div class="content-window">
    <div class="scrollable">
      <div class="browser">
        <section class="finder-container">
          <header class="finder-header">Folders</header>
          <TipeFinder-stub files="" layout="grid" icon="rectangle"></TipeFinder-stub>
        </section>
        <section class="finder-container">
          <header class="finder-header">Documents</header>
          <TipeFinder-stub files="" layout="grid" icon="rectangle"></TipeFinder-stub>
        </section>
      </div>
    </div>
  </div>
</section>
`;
